packages/compass-components/src/components/document-list/element-editors.tsx

```diff
@@ -0,0 +1,336 @@
+    boxShadow: `0 0 0 2px ${uiColors.focus}`,
+  },
+});
+
+const editorInvalid = css({
+  backgroundColor: uiColors.red.light2,
+  color: uiColors.red.dark2,
+  '&:focus, &:active': {
+    boxShadow: `0 0 0 2px ${uiColors.red.dark2}`,
+  },
+});
+
+export const KeyEditor: React.FunctionComponent<{
+  editing?: boolean;
+  onEditStart(): void;
+  value: string;
+  valid: boolean;
+  validationMessage: string | null;
+  onChange(newVal: string): void;
+  autoFocus?: boolean;
+}> = ({
+  editing,
+  value,
+  valid,
+  validationMessage,
+  onChange,
+  autoFocus,
+  onEditStart,
+}) => {
+  const width = `${Math.max(value.length, 1)}ch`;
+
+  return (
+    <>
+      {editing ? (
+        <Tooltip
+          darkMode
+          isDisabled={valid}
+          delay={600}
+          usePortal={false}
+          trigger={({
+            className,
+            children,
+            // Having a tooltip connected to the input elements is not the most
+            // accessible thing ever and so a lot of event listeners of the
+            // tooltip conflict with the textarea default behavior (due to
+            // preventDefault). Because of that we exclude them, so the tooltip
+            // will still be visible, but only on hover or focus, which is okay
+            // for our case
+            /* eslint-disable @typescript-eslint/no-unused-vars */
+            onDragStart,
+            onPointerUp,
+            onPointerDown,
+            onMouseDown,
+            /* eslint-enable @typescript-eslint/no-unused-vars */
+            ...triggerProps
+          }: React.HTMLProps<HTMLInputElement>) => {
+            return (
+              <div className={className}>
+                <input
+                  type="text"
+                  value={value}
+                  onChange={(evt) => {
+                    onChange(evt.currentTarget.value);
+                  }}
+                  // eslint-disable-next-line jsx-a11y/no-autofocus
+                  autoFocus={autoFocus}
+                  className={cx(
+                    maxWidth,
+                    editorReset,
+                    editorOutline,
+                    !valid && editorInvalid
+                  )}
+                  style={{ width }}
+                  spellCheck="false"
+                  {...triggerProps}
+                ></input>
+                {children}
+              </div>
+            );
+          }}
+        >
+          {validationMessage}
+        </Tooltip>
+      ) : (
+        // Double-click is not accessible so no reason for this to be a button
+        <div onDoubleClick={onEditStart} className={maxWidth} style={{ width }}>
+          {value}
+        </div>
+      )}
+    </>
+  );
+};
+
+const textareaContainer = css({
+  width: '100%',
+  maxWidth: '100%',
+  '&::before, &::after': {
+    content: "'\"'",
+    userSelect: 'none',
+  },
+});
+
+const editorTextarea = css({
+  display: 'inline-block',
+  whiteSpace: 'nowrap',
+  minWidth: '3ch',
```

Comment:
I was finding clicking on the empty string input to be a little challenging as the expand handle takes up a good amount of the space when there isn't any string. How do you feel about a few characters extra here?
```suggestion
  minWidth: '5ch',
```
Before:
<img width="271" alt="Screen Shot 2022-03-25 at 11 55 38 AM" src="https://user-images.githubusercontent.com/1791149/160156417-6b2ef82b-b96c-4a5a-aea1-fb8ae362757f.png">

After:
<img width="215" alt="Screen Shot 2022-03-25 at 11 55 03 AM" src="https://user-images.githubusercontent.com/1791149/160156428-aed2f7bb-3446-4ed7-a7ac-9d0edef52f11.png">

Reply:
Oh, totally, I just checked that even before it was 5ch, I just messed it up when re-implementing. Thanks for spotting!